### PR TITLE
Add section on securing service app with an admin role

### PR DESCRIPTION
**Pull request by mauricesharp-okta**
## Latest preview build
Jan 3, 2024 @ 3:00 EDT
https://6595bc18a5d8a20d8397ed9f--reverent-murdock-829d24.netlify.app/docs/guides/terraform-enable-org-access/main/#assign-admin-roles

## Description:

Added a section on adding admin roles to the service app used by Terraform.

### Resolves:

* [OKTA-654929](https://oktainc.atlassian.net/browse/OKTA-654929)

### CODE REVIEW

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx Scorecards

Acrolinx checked the content changed here.

Please review the scorecards for each file and make the recommended
improvements.

| Article | Score | Issues | Scorecard | Status |
| ------- | ----- | ------ | --------- | ------ |
| [packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md](https://github.com/okta/okta-developer-docs/blob/dc458cb3df266a2ede4fdfd6b1e696829e053f5c/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Fguides%2Fterraform-enable-org-access%2Fmain%2Findex.md) | 84 | 33 | [→](https://okta.acrolinx.cloud/api/v1/checking/scorecards/0069dab3-31a7-4fb1-a090-55c52796aecb) | [:white_check_mark:](# "Successfully processed") |

Successfully checked 1 of 1 documents.
See summary in [Content Analysis Dashboard](https://okta.acrolinx.cloud/api/batch/gen.github.5b65d867-a22f-4dd2-8a84-045e093c3216?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

Reopen the pull request or push new changes to check again.

Depending on the Acrolinx server configuration, the
links expire after some time and you must have a login for the
Acrolinx server to access them again.

**Review by monde — APPROVED** [review]

**Comment by monde** [comment]
@mauricesharp-okta it looks like one of of the code owners with a -okta name will have to approve this for merging. Looks good to me. This replaces #4377 correct?

File: packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md [hunk 1]
```diff
@@ -91,6 +91,19 @@ If the key isn't in the right format, convert it to the correct format using the
 
 > **Note:** The file that contains the converted private key must begin with `-----BEGIN RSA PRIVATE KEY-----`. If not, try step two again.
 
+## Assign admin roles
+
+Limit the access of the service app to your org by assigning it one or more admin roles. This increases the security of your org by defining which resources are accessible.
+
+1. In the Admin Console, open the service app and select **Admin roles**.
```
Comment (danielmaharry-okta) [inline packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md:98]: Probably needs a step to get to the apps page, so something like this:

```suggestion
1. In the Admin Console, go to **Applications** > **Applications** to view the current app integrations
1. Select the service app, and then select **Admin roles**.
```

File: packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md [hunk 2]
```diff
@@ -91,6 +91,19 @@ If the key isn't in the right format, convert it to the correct format using the
...
+2. Click **Edit assignments**.
+3. In the **Complete the assignment** section, for **Role** select **Group Administrator**.
+4. Click **Save Changes**.
+
+> **Note:** The example Terraform configuration in this guide requires only the `Group Administrator` role. Create an appropriate custom role your own service app.
```
Comment (danielmaharry-okta) [inline packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md:103]:
```suggestion
> **Note:** The example Terraform configuration in this guide requires only the `Group Administrator` role. Create an appropriate custom role for your service app.
```

**Review by danielmaharry-okta — APPROVED** [review]
Couple of changes to make.
Pre-approving to keep things going.

File: packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md (hunk 2 above)
Comment (mauricesharp-okta) [reply packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md:103]: Changed. In next commit.

File: packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md (hunk 1 above)
Comment (mauricesharp-okta) [reply packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md:98]: This is an "I agree, but then it's inconsistent with the other instances of steps in the article". If you look at the other steps in this article you'll see a pattern. The first set in the Overview gives explicit directions starting with selecting Applications > Applications. The next set in that section refers to opening the service app "create above."

In the next section (Create access credentials), the first step is structured just like the one I added in the new section.

The implication is that the person is going through the whole article so doesn't need the extra step. If that's not the case, then if we make this change, we should also change the one in Create access credentials. I don't care which.

Do you still want the change?

Comment (danielmaharry-okta) [reply packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md:98]: No. Let's stay consistent with the rest of the article. I was looking only at your changes so didn't read it in context.

**Review by MarkCowan-okta — APPROVED** [review]
Lgtm. 🚢

File: packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md [hunk 3]
```diff
@@ -91,6 +91,19 @@ If the key isn't in the right format, convert it to the correct format using the
 
 > **Note:** The file that contains the converted private key must begin with `-----BEGIN RSA PRIVATE KEY-----`. If not, try step two again.
 
+## Assign admin roles
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md:94]: @mauricesharp-okta , @MikeMondragon-okta: this feature doesn't apply to all orgs. You can confirm with @brentschaus-okta and the ACE team.

What orgs are terraform users using for this example? Maybe you can add a note in the beginning, like:

> **Note:** This step is only required for Preview orgs. For Production orgs, this step is required only if the **Assign admin roles to public client apps** early access feature is enabled.

File: packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md [hunk 4]
```diff
@@ -91,6 +91,19 @@ If the key isn't in the right format, convert it to the correct format using the
...
+
+Limit the access of the service app to your org by assigning it one or more admin roles. This increases the security of your org by defining which resources are accessible.
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md:96]: The service app initially has no access. These steps add resource access.

File: packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md [hunk 5]
```diff
@@ -91,6 +91,19 @@ If the key isn't in the right format, convert it to the correct format using the
...
+
+1. In the Admin Console, open the service app and select **Admin roles**.
+2. Click **Edit assignments**.
+3. In the **Complete the assignment** section, for **Role** select **Group Administrator**.
+4. Click **Save Changes**.
+
+> **Note:** The example Terraform configuration in this guide requires only the `Group Administrator` role. Create an appropriate custom role your own service app.
+
+For more information on admin roles, see [Assign admin roles to the OAuth 2.0 service app](/docs/guides/secure-oauth-between-orgs/main/#assign-admin-roles-to-the-oauth-2-0-service-app).
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md:105]: You should link to the generic doc (the one linked here is a specific use case):

https://developer.okta.com/docs/guides/implement-oauth-for-okta-serviceapp/main/#assign-admin-roles-to-the-oauth-2-0-service-app

**Review by vanngo-okta — COMMENTED** [review]
@mauricesharp-okta , @MikeMondragon-okta : some comments for you about this feature. I'm not sure what orgs are used for the terraform example, but we need a note to inform users about when to apply these instructions. Users won't see the option if the FF isn't enabled in production orgs.

**Pull request merged**